packages/lib-classifier/src/store/SubjectStore/Subject/Subject.js

```diff
@@ -67,7 +66,16 @@ const Subject = types
 
         if (!viewer && counts.total > 1 && counts.total < 11) {
           if (!nullViewer) {
-            viewer = subjectViewers.multiFrame
+            if (counts.total === counts.images) {
+              viewer = subjectViewers.multiFrame
+            }
+            if (
+              counts.total === 2 &&
+              counts.images === 1 &&
+              counts.text === 1
+            ) {
+              viewer = subjectViewers.imageAndText
+            }
```

Comment:
Can we reflect on the current model with `getType(self)` to determine which type of subject this is, and thus which viewer to use? That would save having the refinement logic in two places eg. this will break if the refinement is edited but the Subject model isn't updated.
https://mobx-state-tree.js.org/API/#gettype

Reply:
Maybe
```js
if (getType(self) === ImageAndTextSubject) {
  viewer = subjectViewers.imageAndText
}
```

`ImageAndTextSubject` then guarantees that the conditions are met for number of images and number of text files.

Reply:
yes the duplicative refinement logic certainly not ideal, refactored accordingly and I think this is better